docs/contracts.rst

```diff
@@ -209,89 +330,33 @@ Each Contract Factory exposes the following methods.
-
-    * ``filter``: ``dictionary`` - (optional) Dictionary keys should be
-      argument names for the Event arguments.  Dictionary values should be the
-      value you want to filter on, or a list of values to be filtered on.
-      Lists of values will match log entries who's argument matches any value
-      in the list.
-    * ``fromBlock``: ``integer/tag`` - (optional, default: "latest") Integer
-      block number, or "latest" for the last mined block or "pending",
-      "earliest" for not yet mined transactions.
-    * ``toBlock``: ``integer/tag`` - (optional, default: "latest") Integer
-      block number, or "latest" for the last mined block or "pending",
-      "earliest" for not yet mined transactions.
-    * ``address``: ``string`` or list of ``strings``, each 20 Bytes -
-      (optional) Contract address or a list of addresses from which logs should
-      originate.
-    * ``topics``: list of 32 byte ``strings`` or ``null`` - (optional) Array of
-      topics that should be used for filtering.  Topics are order-dependent.
-      This parameter can also be a list of topic lists in which case filtering
-      will match any of the provided topic arrays.
-
-    The event topic for the event specified by ``event_name`` will be added to
-    the ``filter_params['topics']`` list.
-
-    If the :py:attr:`Contract.address` attribute for this contract is
-    non-null, the contract address will be added to the ``filter_params``.
+.. py:class:: ContractEvents
 
-.. _event-log-object:
-
-Event Log Object
-~~~~~~~~~~~~~~~~
-
-    The Event Log Object is a python dictionary with the following keys:
-
-    * ``args``: Dictionary - The arguments coming from the event.
-    * ``event``: String - The event name.
-    * ``logIndex``: Number - integer of the log index position in the block.
-    * ``transactionIndex``: Number - integer of the transactions index position
-      log was created from.
-    * ``transactionHash``: String, 32 Bytes - hash of the transactions this log
-      was created from.
-    * ``address``: String, 32 Bytes - address from which this log originated.
-    * ``blockHash``: String, 32 Bytes - hash of the block where this log was
-      in. null when its pending.
-    * ``blockNumber``: Number - the block number where this log was in. null
-      when its pending.
+The named events exposed through the :py:attr:`Contract.events` property are of the ContractEvents type. This class it not to be used directly, but instead through :py:attr:`Contract.events`.
 
+For example:
 
     .. code-block:: python
 
-        >>> transfer_filter = my_token_contract.eventFilter('Transfer', {'filter': {'_from': '0xdc3a9db694bcdd55ebae4a89b22ac6d12b3f0c24'}})
-        >>> transfer_filter.get_new_entries()
-        [...]  # array of Event Log Objects that match the filter.
-
-        # wait a while...
+        myContract = web3.eth.contract(address=contract_address, abi=contract_abi)
+        myContract.events.myEvent().processReceipt()
```

Comment:
This code example seems a bit strange since I thought `processReceipt` expects an instance of a receipt.